File: growing-shop-java/src/main/java/com/example/growingshop/domain/user/repository/UserRepository.java
package com.example.growingshop.domain.user.repository;

import com.example.growingshop.domain.user.domain.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {
    Optional<User> findUsersByLoginId(String loginId);
}

File: growing-shop-auth/src/main/kotlin/com/example/growingshopauth/user/repository/UserRepository.kt
package com.example.growingshopauth.user.repository

import com.example.growingshopauth.user.domain.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface UserRepository : JpaRepository<User, Long> {
    fun findUsersByLoginId(loginId: String): Optional<User>
}
